### PR TITLE
Add skip-manim tag when building the docs

### PR DESCRIPTION
**Pull request by leotrs**
## List of Changes
- This PR allows us to run `make html O=-tskip-manim` in order to build the documentation without rendering manim code examples.

## Motivation
Sometimes when I build the docs, _all_ manim scenes are re-rendered, even if the files they come from have not been modified.

## Explanation for Changes
This PR makes it so that running sphinx with the `-t skip-manim` tag will signal to the manim directive to not render anything. This tag can be used in one of two ways: `sphinx-build -M html <source_dir> <dest_dir> -t skip-manim` OR by using the makefile: `make html O=-tskip-manim`. Unfortunately, ugly as it is, this is the only way to pass the -t tag to the underlying sphinx process.

## Testing Status
Local pass.

## Further Comments
Right now, the output looks ugly. If anyone has any ideas for how to make it prettier, then I'm all ears. However I'm not too concerned with it since the point of the -t skip-manim flag is to be used only for quick-and-dirty things. No need for pretty.

### CODE REVIEW

File: docs/source/manim_directive.py [hunk 1]
```diff
@@ -123,6 +137,13 @@ class ManimDirective(Directive):
     final_argument_whitespace = True
 
     def run(self):
+        if "skip-manim" in self.state.document.settings.env.app.builder.tags.tags:
+            node = skip_manim_node()
+            self.state.nested_parse(
+                StringList(self.content[0]), self.content_offset, node
+            )
+            return [node]
```
Comment (kolibril13) [inline docs/source/manim_directive.py:145]: Is this return value used somewhere?
I can not find it.

**Comment by kolibril13** [comment]
This is a really good and convenient idea!
I like this functionality and think it can be quite useful.
A question about `sphinx-build -M html <source_dir> <dest_dir> -t skip-manim` :
Does this mean, that source_dir is only one .rst file, like e.g. examples.rst and the rest is ignored and not rendered? 
That would be also nice if one works at one specific .rst file to enhance the docs.

**Comment by MysaaJava** [comment]
> This is a really good and convenient idea!
> I like this functionality and think it can be quite useful.
> A question about `sphinx-build -M html <source_dir> <dest_dir> -t skip-manim` :
> Does this mean, that source_dir is only one .rst file, like e.g. examples.rst and the rest is ignored and not rendered? 
> That would be also nice if one works at one specific .rst file to enhance the docs.

It might also be possible to make sphinx not rebuild .rst files that did not change. But i don't know if it is really the spirit of sphinx.
Also, it seems weird that you can specify not to build manim videos. Is it possible to make a hash of the code to render and then use something like the scene caching, by adding a file keeping track of generated scenes.
I hope you understand what i'm thinking of ^^

File: docs/source/manim_directive.py (hunk 1 above)
Comment (leotrs) [reply docs/source/manim_directive.py:145]: This is used by Sphinx at some later point.

**Comment by leotrs** [comment]
> This is a really good and convenient idea!
> I like this functionality and think it can be quite useful.
> A question about `sphinx-build -M html <source_dir> <dest_dir> -t skip-manim` :
> Does this mean, that source_dir is only one .rst file, like e.g. examples.rst and the rest is ignored and not rendered?
> That would be also nice if one works at one specific .rst file to enhance the docs.

Unfortunately, I don't know. In almost all cases, the actual command is `sphinx-build -M html source/ build/ -t skip-manim` (specifying the whole of the source directory).

I agree it would be nice tho, but I don't know how to do it.

**Comment by leotrs** [comment]
> Also, it seems weird that you can specify not to build manim videos. Is it possible to make a hash of the code to render and then use something like the scene caching, by adding a file keeping track of generated scenes.
I hope you understand what i'm thinking of ^^

I agree it's messy, but remember this is just a debugging tool, so I personally have no problem with my debugging being messy :wink:. I agree what you suggest would be great, but perhaps we can leave it for a future PR? This was meant as a quick patch for now.

**Comment by behackl** [comment]
This seems good (and very useful!) to me, I'll just try it out locally before approving this PR. Thanks for your efforts!

**Review by behackl — APPROVED** [review]
This is great!

You are right, the example blocks look a bit weird, but I really wouldn't care about that too much for now. This does precisely what it should and will allow us to find warnings in the docs much quicker than usual.

A pipeline could even run with the `-W` flag which turns warnings into errors and would allow us to detect ill-formatted documentation *extremely*  easy.

Thank you very much for your effort!

**Review by kolibril13 — APPROVED** [review]

**Pull request merged**